### PR TITLE
special notation for initial and final cycle points

### LINKED ISSUE #1038: special notation for initial and final cycle points
(to consider after #996 is merged)

We can currently start (and end) recurrences at the initial (and final) cycle point using our simplified "in suite context" notation, which omits the end points entirely from the appropriate recurrence form.  This works and is well defined *but* is not exactly easy to read at a glance particularly for the "fourth recurrence form" and final cycle point.

Should we consider either requiring or allowing explicit ```^``` and ```$``` characters to represent suite initial and final points in recurrences, instead of omission - that would be easy to read and consistent with our in-graph notation (at least for initial point ... as I recall we didn't see a use case for in-graph final point notation yet).

@benfitzpatrick - thoughts?

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes #1038 

Sections can now be specified like:
- `[[[ R1/^+PT1H ]]]`
- `[[[ R1/$-PT1H ]]]`

@benfitzpatrick Please Review
@hjoliver Please Review

### CODE REVIEW

**Comment by hjoliver** [comment]
This is a nice, simple solution.  We'll need to document it in the User Guide though.  I'll leave this under "next-release" for now, in case this is all done in time, otherwise will push it back to "soon" so we can get the recent rash of bug fixes out quickly...

File: doc/cug.tex [hunk 1]
```diff
@@ -2822,9 +2822,13 @@ \subsubsection{Graph Types}
             graph = "delayed_cold_corge => corge"
 
         # Repeat once at the final cycle point
-        [[[ R1/P0Y ]]]
+        [[[ R1/$ ]]]
             graph = "end_garply"
 
+        # Repeat once three days before the final cycle point
```
Comment (hjoliver) [inline doc/cug.tex]: This is consistent with our existing documentation, but I wonder if we should change "Repeat once" to "Run once" - the former is nonsensical if interpreted correctly (to repeat something once really suggest to doing it twice)??

Comment (hjoliver) [inline doc/cug.tex]: (can be another PR)

File: doc/cug.tex [hunk 2]
```diff
@@ -2822,9 +2822,13 @@ \subsubsection{Graph Types}
             graph = "delayed_cold_corge => corge"
 
         # Repeat once at the final cycle point
-        [[[ R1/P0Y ]]]
+        [[[ R1/$ ]]]
```
Comment (hjoliver) [inline doc/cug.tex]: We should retain documentation for the implicit form as well (as it is still supported).

In fact, for completeness we should probably document the various ways of getting the same result, and maybe point out which recurrence form each conforms to (Rn/INTVL/STOP, Rn/START/INTVL), for the new explicit notation and the existing implicit forms.  In this case:
 * `R1/P0Y`  
 * `R1/P0Y/$`
 * `R1/$`
 
It can get confusing, especially when you consider cases where dropping an explicit `$` gives a different result by turning a point into an interval - `R2/$-P1Y/P1Y` to `R2/-P1Y/P1Y` ...

File: doc/cug.tex (hunk 1 above)
Comment (benfitzpatrick) [inline doc/cug.tex]: :+1:

File: doc/cug.tex (hunk 2 above)
Comment (benfitzpatrick) [inline doc/cug.tex]: If the example list gets too unwieldy with extra additions it might be best to split it up and have it under different sub-sections.

**Comment by benfitzpatrick** [comment]
> This is a nice, simple solution

I agree!

File: doc/cug.tex (hunk 1 above)
Comment (oliver-sanders) [inline doc/cug.tex]: Will do in a future pull request.

**Comment by hjoliver** [comment]
(pushed back to soon; we can bring forward again if done in time).

File: doc/cug.tex (hunk 2 above)
Comment (oliver-sanders) [inline doc/cug.tex]: I've added comments showing which recurrence form the headings conform to in the caret/dollar notation section, I can do this for the main list of examples as well if desired (though the recurrence forms aren't explained until after the list).

File: doc/cug.tex [hunk 3]
```diff
@@ -2940,6 +2948,40 @@ \subsubsection{Graph Types}
...
 
+For convenience the caret and dollar symbols may be used as shorthand for the
+initial and final cycle points. Using this shorthand you can write:
+
+\lstset{language=suiterc}
+\begin{lstlisting}
+[scheduling]
+    [[dependencies]]
+        # Repeat once 12 hours after the initial cycle point
+        # Equivalent to [[[ R1/T12 ]]]
```
Comment (benfitzpatrick) [inline doc/cug.tex]: Equivalent to [[[ R1/+PT12H ]]]

**Comment by benfitzpatrick** [comment]
One comment to fix, otherwise looks good.

**Comment by oliver-sanders** [comment]
@benfitzpatrick Comment addressed.

**Comment by benfitzpatrick** [comment]
0899c38 looks good.
